src/ManageCourses.Api/Controllers/AdminController.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections.ObjectModel;
+using System.Linq;
+using GovUk.Education.ManageCourses.Api.ActionFilters;
+using GovUk.Education.ManageCourses.Api.Middleware;
+using GovUk.Education.ManageCourses.Domain.DatabaseAccess;
+using GovUk.Education.ManageCourses.Domain.Models;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using static GovUk.Education.ManageCourses.Domain.Models.AccessRequest;
+
+namespace GovUk.Education.ManageCourses.Api.Controllers
+{
+    [Route("api/admin")]
+    public class AdminController : Controller
+    {
+        private readonly IManageCoursesDbContext _context;
+
+        public AdminController(IManageCoursesDbContext context)
+        {
+            _context = context;
+        }
+
+        [HttpPost("access-request")]
+        [ApiTokenAuth]
+        [ExemptFromAcceptTerms]
+        public IActionResult ActionAccessRequest(int accessRequestId)
+        {
+            // Include instructs EF to also retrieve data that's connected with a foreign key
+            // With ThenInclude you can also retrieve relations of relations
+            var request = _context.AccessRequests
+                .Include(accessRequest => accessRequest.Requester)
+                .ThenInclude(requester => requester.McOrganisationUsers)
+                .FirstOrDefault(x => x.Id == accessRequestId);
+
+            if (request == null)
+            {
+                // Returns a 404.
+                return NotFound();
+            }
+
+            var requesterEmail = (request.Requester?.Email ?? request.RequesterEmail).ToLower();            
+            var requesterUser = request.Requester ?? _context.McUsers.Include(x => x.McOrganisationUsers).FirstOrDefault(x => x.Email == requesterEmail);
```

Comment:
These two lines of fallbacks make me a bit uneasy. Perhaps a comment explaining the circumstances that led to needing the fallbacks would be useful so that we will be able to know when the time comes to simplify the code whether the reasons have gone away.

minor

Reply:
👍